index.js

```diff
@@ -1,18 +1,74 @@
 const readline = require('readline');
 const rl = readline.createInterface(process.stdin, process.stdout);
-
+let minNum = 1
+let maxNum = 100
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     rl.question(questionText, resolve);
   });
 }
 
+// Original random function
+// function randomNum(min, max) {
+//   return Math.floor(Math.random() * (max - min + 1) + min)
+// }
+
+// binary search function
+function randomNum(min, max) {
+  return Math.floor((min + max) / 2)
+}
+
 start();
 
 async function start() {
   console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
   let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
```

Comment:
You can add a small while loop here that can help see if secretNumber is in fact a number, and if it's not, have it loop through asking for a number until a number is provided.